src/librustc/diagnostics.rs

```diff
@@ -1317,6 +1317,58 @@ let x: i32 = "I am not a number!";
 //      |
 //    type `i32` assigned to variable `x`
 ```
+
+You will get this error too if you want to return a trait (implementor):
```

Comment:
Hmm. I wonder if we should assign this case a distinct error code; it feels like this text is getting awfully long, but also somehow not long enough. In other words, it'd maybe be good to provide a few pointers into the book and give just a *bit* more detail.

Reply:
In particular, I'd maybe want to talk about how the actual meaning of this is to return a "trait object", which is likely not what you expected. Not sure just how to approach that topic in this limited space though.